meerk40t/svgelements.py

```diff
@@ -190,22 +201,56 @@
     + "|"
     + SVG_TRANSFORM_SKEW_Y
 )
-PATTERN_TRANSFORM_UNITS = (
-    PATTERN_LENGTH_UNITS + "|" + PATTERN_ANGLE_UNITS + "|" + PATTERN_PERCENT
+PATTERN_TRANSFORM_UNITS = r"|".join(
+    (PATTERN_LENGTH_UNITS, PATTERN_ANGLE_UNITS, PATTERN_PERCENT)
 )
+TRANSFORM_UNIT_SINGULARS = {
+    "": "",
+    "%": "%",
+    "cm": "cm",
+    "cms": "cm",
+    "mm": "mm",
+    "mms": "mm",
+    "in": "in",
+    "ins": "in",
+    "inch": "in",
+    "inches": "in",
+    "mil": "mil",
+    "mils": "mil",
+    "pt": "pt",
+    "pts": "pt",
+    "deg": "deg",
+    "degs": "deg",
+    "grad": "grad",
+    "grads": "grad",
+    "gradian": "grad",
+    "gradians": "grad",
+    "rad": "rad",
+    "rads": "rad",
+    "radian": "rad",
+    "radians": "rad",
+    "turn": "turn",
+    "turns": "turn",
+}
 
 REGEX_IRI = re.compile(r"url\(#?(.*)\)")
 REGEX_DATA_URL = re.compile(r"^data:([^,]*),(.*)")
-REGEX_FLOAT = re.compile(PATTERN_FLOAT)
+REGEX_FLOAT = re.compile(r"(%s)" % PATTERN_FLOAT)
```

Comment:
REGEX_FLOAT is not purely about units it's just a regex of the floats.

Reply:
Not sure what this comment means - the purpose of this change was to capture the group.